### DISCUSSION TITLE
How to get the mouse cursor position

### DISCUSSION THREAD

**Discussion by ben-cottrell-nz** [slint-ui/slint#2769] [Q&A]
Within a component, I would like to get the mouse position always, not just when dragging. In the below snippet, I only see cursor position when clicking and moving.
```
export component TanksGameView inherits Window {
        width: 1024px;
        height: 768px;
        forward-focus: player-key-handler;
        touchArea := TouchArea {
            width: parent.width;
            height: parent.height;
            moved => {
                debug(touchArea.mouse-x / 1px + "," + touchArea.mouse-y / 1px)
            }
        }
}
```

**Comment by ogoffart** [comment]
This is currently not supported by the TouchArea
Thanks for creating the issue https://github.com/slint-ui/slint/issues/2770

**Comment by Firemanpl** [comment]
so what how is working ripple animation on Button ?

**Comment by tenpaMk2** [comment]
Use the `Timer` and poll `mouse-x` and `mouse-y` periodically (ex: 100ms interval) in the `TouchArea`.
You can detect mouse move by checking `mouse-x` and `mouse-y` with previous polling values.
